### ISSUE TITLE
`AstroHeader.getVal('nonexisting_key')` returns `NaN` instead of `[]`

### ISSUE THREAD

**Issue by EastEriq** [EranOfek/AstroPack#209]
and one could argue that this is the intended behavior, and that changing it would break other existing code (does it?).

However:

- `imProc.io.writeProduct` contains a hardwired call to `Filenames.updateFromObjectInfo(...'AI_CropID_FromHeader',Args.AI_CropID_FromHeader,'AI_Counter_FromHeader',Args.AI_Counter_FromHeader,...)`
- `Filenames.updateFromObjectInfo()` relies on `Filenames.updateIfNotEmpty('Counter',Counter, 'CROPID',CropID, 'Time',JD)`;
- this means that if one creates and populate a Filenames object with `CROPID` and then calls `imProc.io.writeProduct(AI,F)`, but AI.HeaderData does not contain already `CROPID`, the NaN will prevail.

**Comment by EranOfek** [comment]
Yes - this is the default behaviour. You can control via the 'Fill' argument - see help.

**Comment by EastEriq** [comment]
Fill is not an option for code already written relying on getVal
